### PR TITLE
Fixed NPE issue with Metadata calls in SEA

### PR DESCRIPTION
**Pull request by msrathore-db**
## Description

Fixed NPE in metadata calls for SEA.
## Testing

Tested locally
## Additional Notes to the Reviewer

NO_CHANGELOG=true
Fixes [PECOBLR-1372](https://databricks.atlassian.net/browse/PECOBLR-1372)

[PECOBLR-1372]: https://databricks.atlassian.net/browse/PECOBLR-1372?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/com/databricks/jdbc/dbclient/impl/sqlexec/DatabricksMetadataSdkClient.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public DatabricksResultSet listSchemas(
       return metadataResultSetBuilder.getSchemasResult(getResultSet(SQL, session), catalog);
     } catch (SQLException e) {
       if (WildcardUtil.isNullOrWildcard(catalog)
-          && e.getSQLState().equals(PARSE_SYNTAX_ERROR_SQL_STATE)) {
+          && PARSE_SYNTAX_ERROR_SQL_STATE.equals(e.getSQLState())) {
```
Comment (samikshya-db) [inline src/main/java/com/databricks/jdbc/dbclient/impl/sqlexec/DatabricksMetadataSdkClient.java]: can you add a test that checks null too

Comment (samikshya-db) [reply src/main/java/com/databricks/jdbc/dbclient/impl/sqlexec/DatabricksMetadataSdkClient.java]: The idea of a unit test is so that we don't break this in the future - even though it may seem obvious at the moment.

**Review by gopalldb — APPROVED** [review]
